File: app/src/main/java/de/cineaste/android/entity/series/Season.java
package de.cineaste.android.entity.series;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class Season {

    private long id;
    @SerializedName("air_date")
    private Date releaseDate;
    @SerializedName("episode_count")
    private int episodeCount;
    @SerializedName("poster_path")
    private String posterPath;
    @SerializedName("season_number")
    private int seasonNumber;

    private long seriesId;
    private boolean watched;
    private List<Episode> episodes;

    public Season() {
        this.episodes = new ArrayList<>();
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public Date getReleaseDate() {
        return releaseDate;
    }

    public void setReleaseDate(Date releaseDate) {
        this.releaseDate = releaseDate;
    }

    public int getEpisodeCount() {
        return episodeCount;
    }

    public void setEpisodeCount(int episodeCount) {
        this.episodeCount = episodeCount;
    }

    public String getPosterPath() {
        return posterPath;
    }

    public void setPosterPath(String posterPath) {
        this.posterPath = posterPath;
    }

    public int getSeasonNumber() {
        return seasonNumber;
    }

    public void setSeasonNumber(int seasonNumber) {
        this.seasonNumber = seasonNumber;
    }

    public long getSeriesId() {
        return seriesId;
    }

    public void setSeriesId(long seriesId) {
        this.seriesId = seriesId;
    }

    public boolean isWatched() {
        return watched;
    }

    public void setWatched(boolean watched) {
        this.watched = watched;
    }

    public List<Episode> getEpisodes() {
        return episodes;
    }

    public void setEpisodes(List<Episode> episodes) {
        this.episodes = episodes;
    }
}

File: app/src/main/kotlin/de/cineaste/android/entity/series/Season.kt
package de.cineaste.android.entity.series

import com.google.gson.annotations.SerializedName
import java.text.SimpleDateFormat
import java.util.*

data class Season(
    var id: Long = 0,
    @SerializedName("air_date")
    var releaseDate: Date? = null,
    @SerializedName("episode_count")
    var episodeCount: Int = 0,
    @SerializedName("poster_path")
    var posterPath: String? = null,
    @SerializedName("season_number")
    var seasonNumber: Int = 0,
    var seriesId: Long = 0,
    var isWatched: Boolean = false,
    var episodes: List<Episode> = listOf()
)

fun Season.toEntity() =
    SeasonEntity(id, releaseDate?.let {
        val simpleDateFormat = SimpleDateFormat("dd.MM.yyyy", Locale.ENGLISH)
        simpleDateFormat.format(it)
    }, episodeCount, posterPath, seasonNumber, seriesId, isWatched)